.github/workflows/dependabot-changenote.yml

```
@@ -0,0 +1,45 @@
+name: Dependabot Change Note
+on:
+  push:
+    branches:
+      - 'dependabot/**'
+
+permissions:
+  pull-requests: write
+
+jobs:
+  changenote:
+    name: Dependabot Change Note
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    if: github.actor == 'dependabot[bot]'
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3.1.0
+        with:
+          fetch-depth: 1
+          token: ${{ secrets.BRUTUS_PAT_TOKEN }}
+
+      - name: Configure git
+        run: |
+          git config --local user.email "$(git log --pretty='%ae' -1)"
+          git config --local user.name "Dependabot[bot]"
+
+      - name: Commit Change Note
+        run: |
+          # Fetch PR number for commit from Github API
+          API_URL="${{ github.api_url }}/repos/${{ github.repository }}/commits/${{ github.event.head_commit.id }}/pulls"
+          PR_NUM=$(curl -s -H "Accept: application/vnd.github+json" "${API_URL}" | jq -r '.[].number')
+
+          # Create change note from first line of dependabot commit
+          NEWS=$(printf "${{ github.event.head_commit.message }}" | head -n1 | sed -e 's/Bump/Updated/')
+          printf "${NEWS}.\n" > "./changes/${PR_NUM}.misc.rst"
```

This will evaluate `printf` format specifications and escape sequences in the commit message (twice!), which probably isn't intended. The safe version would be:

```suggestion
          NEWS=$(printf "%s" "${{ github.event.head_commit.message }}" | head -n1 | sed -e 's/Bump/Updated/')
          printf "%s.\n" "${NEWS}" > "./changes/${PR_NUM}.misc.rst"
```

It would be nice to get rid of the double `printf` too, but I can't think of an easy way to do that without breaking the line splitting. I'm sure there's a good `sed` or `awk` one-liner for this, but I don't know how what it would be.

---

True - but it's also calling `head -n1`, and the first line of the message won't (can't?) ever have anything that will be risky from a formatting perspective. You can see commits like [this one](https://github.com/beeware/briefcase/pull/959/commits/13f779c60820a2d1fb27c1c9ae40b6d9aed8d967) for what gets generated.

---

The `head -n1` is after the first `printf` call, so even formatting characters *anywhere* in the message could cause problems. You're right that it's unlikely to be an issue since the commit message is auto-generated and has a fixed format - but since the code is already using `printf` anyway, is there any reason to *not* do it the safe way?

---

The main reason is inertia :-) . This config is the same as the one that is used on Toga and Briefcase, where it's been working well; it's a piece of code that is exceedingly difficult to test in it's entirely; and if something breaks, it's a PITA to fix it. 

However, none of those are especially *good* reasons; I'll put updating this on my list as a TODO.